fish-rust/src/trace.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    common::{escape_string, EscapeFlags},
+    ffi::{self, parser_t, wcharz_t},
+    wchar::{self, wstr, L},
+    wchar_ffi::WCharToFFI,
+};
+use std::sync::atomic;
+
+#[cxx::bridge]
+mod ffi2 {
+    extern "C++" {
+        include!("wutil.h");
+        include!("parser.h");
+        type wcharz_t = super::wcharz_t;
+        type parser_t = super::parser_t;
+    }
+
+    extern "Rust" {
+        fn rust_trace_set_enabled(do_enable: bool);
+        fn rust_trace_enabled(parser: &parser_t) -> bool;
+        fn rust_trace_argv(parser: &parser_t, command: wcharz_t, args: &Vec<wcharz_t>);
+        fn rust_trace_if_enabled(parser: &parser_t, command: wcharz_t);
+    }
+}
+
+static do_trace: atomic::AtomicBool = atomic::AtomicBool::new(false);
+
+fn rust_trace_set_enabled(do_enable: bool) {
+    do_trace.store(do_enable, atomic::Ordering::Relaxed);
+}
+
+fn rust_trace_enabled(parser: &parser_t) -> bool {
+    let ld = parser.ffi_libdata_pod_const();
+    if ld.suppress_fish_trace {
+        return false;
+    }
+    return do_trace.load(atomic::Ordering::Relaxed);
+}
+
+/// Trace an "argv": a list of arguments where the first is the command.
+/// Allow the `&Vec` parameter as this function only exists temporarily for the FFI
```

Comment:
Nitpick: The second comment should be a regular `//` and not a doc comment (`///`).

Reply:
Done.